src/index.ts

```diff
@@ -287,54 +286,76 @@ const extension: JupyterFrontEndPlugin<void> = {
+      if (!enableCloudStorage) {
         panelGcs?.dispose();
         gcsDrive?.dispose();
-        panelDpms = undefined;
-        panelDatasetExplorer = undefined;
         panelGcs = undefined;
         gcsDrive = undefined;
-      } else {
-        // Preview was enabled, (re)create DPMS and GCS.
-        if (!panelDpms && !panelGcs) {
-          panelDpms = new Panel();
-          panelDpms.id = 'dpms-tab';
-          panelDpms.title.caption = 'Dataset Explorer - DPMS';
-          panelDpms.addWidget(new dpmsWidget(app as JupyterLab, themeManager));
-          if (bqFeature.enable_bigquery_integration && !panelDatasetExplorer) {
-            panelDatasetExplorer = new Panel();
-            panelDatasetExplorer.id = 'dataset-explorer-tab';
-            panelDatasetExplorer.title.caption = 'Dataset Explorer - BigQuery';
-            panelDatasetExplorer.addWidget(
-              new BigQueryWidget(
-                app as JupyterLab,
-                settingRegistry as ISettingRegistry,
-                bqFeature.enable_bigquery_integration as boolean,
-                themeManager
-              )
-            );
-          }
-          panelGcs = new Panel();
-          panelGcs.id = 'GCS-bucket-tab';
-          panelGcs.title.caption = 'Google Cloud Storage';
-          gcsDrive = new GCSDrive();
-          documentManager.services.contents.addDrive(gcsDrive);
-          panelGcs.addWidget(
-            new GcsBrowserWidget(gcsDrive, factory as IFileBrowserFactory)
-          );
-          // Update the icons.
-          onThemeChanged();
-          app.shell.add(panelGcs, 'left', { rank: 1002 });
-          if (bqFeature.enable_bigquery_integration && panelDatasetExplorer) {
-            app.shell.add(panelDatasetExplorer, 'left', { rank: 1000 });
-          }
+      }
+      if (!enableBigQuery) {
+        panelDatasetExplorer?.dispose();
+        panelDatasetExplorer = undefined;
+      }
+      if (
+        bqFeature.enable_bigquery_integration ||
+        bqFeature.enable_cloud_storage_integration ||
+        bqFeature.enable_metastore_integration
+      ) {
+        panelDpms = new Panel();
+        panelDpms.id = 'dpms-tab';
+        panelDpms.title.caption = 'Dataset Explorer - DPMS';
+        panelDpms.addWidget(new dpmsWidget(app as JupyterLab, themeManager));
+        panelDatasetExplorer = new Panel();
+        panelDatasetExplorer.id = 'dataset-explorer-tab';
+        panelDatasetExplorer.title.caption = 'Dataset Explorer - BigQuery';
+        panelDatasetExplorer.addWidget(
+          new BigQueryWidget(
+            app as JupyterLab,
+            settingRegistry as ISettingRegistry,
+            bqFeature.enable_bigquery_integration as boolean,
+            themeManager
+          )
+        );
+        panelGcs = new Panel();
+        panelGcs.id = 'GCS-bucket-tab';
+        panelGcs.title.caption = 'Google Cloud Storage';
+        gcsDrive = new GCSDrive();
+        documentManager.services.contents.addDrive(gcsDrive);
```

Comment:
We cannot do this if the GCS extension is disabled. If it is disabled then *all* support for GCS needs to be disabled.